netbox_netprod_importer/vendors/cisco/ios.py

```diff
@@ -78,3 +107,138 @@ def get_detailed_cdp_neighbours(self):
                 )
         for n in neighbours:
             yield n
+
+    def get_interface_mode(self, interface):
+        from pynxos.errors import CLIError
+        try:
+            return self._get_interfaces_mode()[
+                self.get_abrev_if(interface)].get("oper_mode")
+        except (KeyError, CLIError):
+            logger.debug("Switch %s, show interface switchport cmd error",
+                         self.device.hostname)
+        return None
```

Comment:
Usually, I prefer raising the exception if no mode was found (and catch it where I called the function) than returning `None`. This way you know it failed.

As you wish, you can let it this way if you prefer.

Reply:
This is a normal situation if the interface is not in the `switchport` but in the `router`.